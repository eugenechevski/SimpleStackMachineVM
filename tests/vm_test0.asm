	# $Id: vm_test0.asm,v 1.3 2024/07/26 12:44:46 leavens Exp $
	.text start
start:	STRA
	ADDI $sp, -1, 2
	EXIT 0
	.data 1024
	.stack 4096
	.end
	
