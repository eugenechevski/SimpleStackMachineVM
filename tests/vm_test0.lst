Address Instruction
     0: STRA 
     1: ADDI $sp, -1, 2
     2: EXIT 0
    1024: 0	        ...     
