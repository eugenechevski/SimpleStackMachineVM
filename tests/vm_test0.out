      PC: 0
GPR[$gp]: 1024 	GPR[$sp]: 4096 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 0	        ...     
    4096: 0	

==>      0: STRA 
      PC: 1
GPR[$gp]: 1024 	GPR[$sp]: 4096 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 0	        ...     
    4096: 0	

==>      1: ADDI $sp, -1, 2
      PC: 2
GPR[$gp]: 1024 	GPR[$sp]: 4096 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 0	        ...         4095: 2	
    4096: 0	

==>      2: EXIT 0
